### DISCUSSION TITLE
How to adapt useEffect call on root layout?

### DISCUSSION THREAD

**Discussion by gustavopolonio** [vercel/next.js#57226] [App Router]
I want to execute an useEffect call on all pages of my site. Before the Next v13 I called this effect on the `_app.ts`. On Next v13 the root layout replaces the `_app.ts`, but it must be server component (therefore we can't use the useEffect hook inside it).

So how can I execute an useEffect on all pages of my site (without execute it separately on each page)?

I thought about create a context and execute the useEffect inside it. But this isn't the best way, since the context api was not created for this purpose. Is there a better solution?

**Comment by sinhapaurush** [comment]
I suppose you will need to use client components for that, if you want to avoid using client components you can create another component which is client component as below
```
'use client' 
import React from 'react';
function SomeComp(){
React.useEffect(()=>{
//DO STUFF 
});
return null;
} 
```
And use it in your root layout, ie in layout.js file as below

```
function Layout({children}) {
return(
<SomeComp />
{children}
);
} 
```
